SIPs/sip_001.md

```diff
@@ -0,0 +1,172 @@
+# SPOOL Improvement Proposal 001
+
+
+## Abstract
+
+This proposal discusses details for allowing the implementation
+of unlimited editions in the SPOOL ownership API.
+
+## Requirements
+
+    - An API user must be able to bulk create a set of unlimited editions for a piece in a single
+      HTTP request
+    - An API user must be able to repeatedly create a set of unlimited editions
+      for a piece using HTTP requests
+    - An API user is not required to know about concurrent edition creation
```

Comment:
> How does one know what's the current amount of editions?
> Imagine that someone is creating 100 editions, will the number of editions update instantaneously (ie before all are generated) or will the number of editions update during generation?

As we cannot break the API at this point, the encoding:

```
num_editions == -1 := no editions created yet
num_editions == 0 := editions are being created
num_editions > 0 && first_edition != null := editions have been created
num_editions > 0 && first_edition == null := editions have been created and all deleted
```

> How will a user (ie the ascribe frontend) know that a background process is running?

Check if num_editions == 0

Obviously, this will be taken care of in v2 of the API...